https://marc.info/?l=linux-netdev&m=151554000532498&w=2

From aec69bbc285e5498add5384b6018564355594233 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@chromium.org>
Date: Tue, 9 Jan 2018 18:16:53 -0500
Subject: [PATCH iproute2] ifcfg/rtpr: convert to POSIX shell

These files are already mostly written in POSIX shell, so convert their
shebangs to /bin/sh and tweak the few bashisms in here.

URL: https://crbug.com/756559
Reported-by: Pat Erley <perley@chromium.org>
Signed-off-by: Mike Frysinger <vapier@chromium.org>
---
 ip/ifcfg | 12 ++++++------
 ip/rtpr  |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/ip/ifcfg b/ip/ifcfg
index 851b9215ab63..5b34decd4480 100644
--- a/ip/ifcfg
+++ b/ip/ifcfg
@@ -1,3 +1,3 @@
-#! /bin/bash
+#! /bin/sh
 
 CheckForwarding () {
@@ -7,7 +7,7 @@ CheckForwarding () {
   fwd=0
   if [ -d $sbase ]; then
     for dir in $sbase/*/forwarding; do
-      fwd=$[$fwd + `cat $dir`]
+      fwd=$(( fwd + $(cat "$dir") ))
     done
   else
     fwd=2
@@ -128,12 +128,12 @@ fi
 arping -q -A -c 1 -I $dev $ipaddr
 noarp=$?
 ( sleep 2 ;
-  arping -q -U -c 1 -I $dev $ipaddr ) >& /dev/null </dev/null &
+  arping -q -U -c 1 -I $dev $ipaddr ) >/dev/null 2>&1 </dev/null &
 
-ip route add unreachable 224.0.0.0/24 >& /dev/null
-ip route add unreachable 255.255.255.255 >& /dev/null
+ip route add unreachable 224.0.0.0/24 >/dev/null 2>&1
+ip route add unreachable 255.255.255.255 >/dev/null 2>&1
 if [ "`ip link ls $dev | grep -c MULTICAST`" -ge 1 ]; then
-  ip route add 224.0.0.0/4 dev $dev scope global >& /dev/null
+  ip route add 224.0.0.0/4 dev $dev scope global >/dev/null 2>&1
 fi
 
 if [ $fwd -eq 0 ]; then
diff --git a/ip/rtpr b/ip/rtpr
index 192a476f817e..7e48674bcf53 100644
--- a/ip/rtpr
+++ b/ip/rtpr
@@ -1,3 +1,3 @@
-#! /bin/bash
+#! /bin/sh
 
 exec tr "[\\\\]" "[
-- 
2.15.1

